plugins/SpectrumAnalyzer/SaProcessor.h

```diff
@@ -102,9 +122,13 @@ class SaProcessor
 	std::vector<float> m_normSpectrumR;     //!< frequency domain samples (normalized) (right)
 
 	// spectrum history for waterfall: new normSpectrum lines are added on top
-	std::vector<uchar> m_history;
-	const unsigned int m_waterfallHeight = 200;	// Number of stored lines.
-												// Note: high values may make it harder to see transients.
+	std::vector<uchar> m_history_work;		//!< local history buffer for render
```

Comment:
`m_history_work` is local "work buffer" for the FFT thread, `m_history` is the display buffer that is used for drawing.
The waterfall widget only needs to access `m_history_work` once: to clear its content when the waterfall window is hidden and shown again. Otherwise it would contain a mix of old and new data, since the waterfall processing is paused when waterfall is hidden.

I could make a `clear_waterfall()` function to do the same thing in SaProcessor, it just seemed a bit unnecessary at the time to make a function that would be used only once in the entire code base. But it would make a better interface, true.

I will take a look at the "friend list" and try to get rid of some of them (where possible without too many changes required in the existing code).

Reply:
OK, I'll leave this open until I'm through with your new commit.